aws-amplify/amplify-ui
# fix(storage-browser): update useHandleUpload to return handleFileSelect function

#### Description of changes
- Update `useHandleUpload` to return a `handleFileSelect` function that helps create and manages previous tasks
- Pass the `handleFileSelect` function to `useFileSelect` so that `handleFileSelect` can accept the new files from the selection and manages the state of these files and tasks
- Disable "start", "add files" and "add folder" buttons after start action is pressed


To test:
1. Try uploading a folder (with multiple files) from the location items table
   - Verify that you see the folder name in the `folder` column
2. Remove one of the files from the folder you uploaded from step 1
   - Verify the file gets removed and nothing gets added to the tasks table
3. Add a new file with the "Add files" button
   - Verify the file gets added
4. Try uploading the same folder you uploaded from step 1 using the "Add folder" button
   - Verify that the folder gets added and the items that were added appear at the top of the table
   
Feel free to test out different combinations of adding files and deleting them and re-adding files



#### Issue #, if available



#### Description of how you validated changes

#### Checklist



- [ ] PR title and commit messages follow [conventional commit](https://www.conventionalcommits.org/en/v1.0.0/#summary) syntax
- [ ] _If this change should result in a version bump_, [changeset added](https://github.com/changesets/changesets/blob/main/docs/adding-a-changeset.md) (This can be done after creating the PR.) This does not apply to changes made to `docs`, `e2e`, `examples`, or other private packages.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
